src/coreclr/vm/dllimport.cpp

```diff
@@ -3615,26 +3643,48 @@ static void CreateNDirectStubWorker(StubState*               pss,
                  &pSigDesc->m_typeContext);
 
     if (SF_IsVarArgStub(dwStubFlags))
+    {
+        if (!builtInMarshallingEnabled)
+        {
+            COMPlusThrow(kMarshalDirectiveException, IDS_EE_NDIRECT_DISABLEDMARSHAL_VARARGS);
+        }
         msig.SetTreatAsVarArg();
+    }
 
     bool fThisCall = (unmgdCallConv == CorInfoCallConvExtension::Thiscall);
 
-    pss->SetLastError(nlFlags & nlfLastError);
+    if (nlFlags & nlfLastError)
+    {
+        if (!builtInMarshallingEnabled)
+        {
+            COMPlusThrow(kMarshalDirectiveException, IDS_EE_NDIRECT_DISABLEDMARSHAL_SETLASTERROR);
+        }
+        pss->SetLastError(TRUE);
+    }
 
     // This has been in the product since forward P/Invoke via delegates was
     // introduced. It's wrong, but please keep it for backward compatibility.
-    if (SF_IsDelegateStub(dwStubFlags))
+    if (builtInMarshallingEnabled && SF_IsDelegateStub(dwStubFlags))
         pss->SetLastError(TRUE);
 
     pss->BeginEmit(dwStubFlags);
 
     if (-1 != iLCIDArg)
     {
-        // The code to handle the LCID  will call MarshalLCID before calling MarshalArgument
+        if (!builtInMarshallingEnabled)
```

Comment:
We silently ignore all other marshaling directives outside DllImport attribute. Should we silently ignore `LCIDConversionAttribute` as well?

Reply:
I think that we should have a predictable failure path for `LCIDConversionAttribute` as it effectively changes the calling convention and would cause very difficult-to-diagnose failures if someone is trying to use it and it's silently disabled.

If we add a Roslyn analyzer to identify the cases that aren't supported when runtime marshalling is disabled and warn the user, I'd be more comfortable removing this check, but I'm not comfortable with a silent failure.